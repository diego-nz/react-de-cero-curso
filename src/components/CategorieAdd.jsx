import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const CategorieAdd = ( {setCategories} ) => {
    
    const [inputValue, setInputValue] = useState('');

    const handleInput = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 0){
            // Accede a las categorias mediante callback para no pasarlas como props
            setCategories( cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <h2>Add categorie</h2>
            <input type="text" value={inputValue} onChange={handleInput}/>
        </form>
    )
}

CategorieAdd.propTypes = {
    setCategories: PropTypes.func.isRequired
}
