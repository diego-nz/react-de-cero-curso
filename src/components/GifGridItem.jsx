import React from 'react';
import PropTypes from 'prop-types'

export const GifGridItem = ( { image, title } ) => {
    return (
        <figure className="card__item">
            <img src={image} alt={title} />
            <figcaption>{title}</figcaption>
        </figure>
    )
}
// import proptypes imrpcp
// Props requeridas
GifGridItem.propTypes = {
    image: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
}
