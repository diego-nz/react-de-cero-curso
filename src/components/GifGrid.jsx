import React from 'react'
// import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'


export const GifGrid = ( { categorie } ) => {
    
    // const [images, setImages] = useState([]);

    // Con : se esta renombrando el key en la destructuracion
    const { data: images, loading } = useFetchGifs( categorie );
    // useEffect(() => {
    //     getGifs( categorie ).then( data => setImages( data ));
    // }, [categorie])


    return (
        <>
            <h3> {categorie} </h3>
            { loading && <h4 className="animate__animated animate__flash">Cargando...</h4> }
            <div className="card-container animate__animated animate__zoomInUp">
                {images.map((img) => 
                    <GifGridItem key={img.id} {...img}/>
                )}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    categorie: PropTypes.string.isRequired
}
