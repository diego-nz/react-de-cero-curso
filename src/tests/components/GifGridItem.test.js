import React from 'react'
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en componente GifGridItem', () => {
    // Para simular las proptypes de gifgriditem
    const title = 'Un titulo';
    const image = 'https://urldemimiagen.com'

    // Para hacer el snapshot y con u en terminal se hace un update si cambia algo en el componente
    const wrapper = shallow(<GifGridItem title={ title } image={ image } />);

    it('debe de mostrar correctamente el componente GifGridItem', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    it('debe de tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find('figcaption');
        expect(p.text()).toBe( title );
    });

    it('debe tener la imagen igual al src asi como alt', () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe(image);
        expect( img.prop('alt') ).toBe(title);
    });
    
    it('debe tener las clases card__item', () => {
        const figure = wrapper.find('figure');
        expect( figure.hasClass('card__item') ).toBe(true);
    });
})
