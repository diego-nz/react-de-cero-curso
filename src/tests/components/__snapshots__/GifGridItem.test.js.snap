// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGridItem debe de mostrar correctamente el componente GifGridItem 1`] = `
<figure
  className="card__item"
>
  <img
    alt="Un titulo"
    src="https://urldemimiagen.com"
  />
  <figcaption>
    Un titulo
  </figcaption>
</figure>
`;
